content/_includes/talks.macros.njk

```diff
@@ -0,0 +1,202 @@
+    default: true
+    note: |
+      If we're only showing upcoming events,
+      we'll reverse the order so that
+      the soonest event is first.
+  heading_level:
+    type: integer (1-6)
+    default: 4
+    note: |
+      Adjust the heading level
+      for individual event titles,
+      depending on the context where this list is shown.
+  compact:
+    type: boolean
+    default: false
+    note: |
+      When showing an event list
+      inside a grid of talks or posts,
+      we only show the most recent 2 events,
+      and simplify the styling/output.
+#}
+{% macro list(
+  events,
+  all=true,
+  heading_level=4,
+  compact=false
+) %}
+  {%- if events and (events | length > 0) -%}
+    {%- set events = events | sort(all, false, 'date') -%}
+    {%- set events = events | onlyShow(2) if compact else events -%}
+
+    <ol {{ utility.show_attrs({
+      'reversed': all,
+      'class': ['event-list', 'compact-events'] | join(' ') if compact else 'event-list'
+    }) }}>
+      {%- for event in events -%}
+        <li class="event-item">
+          {{ meta(event) }}
+
+          {% h heading_level, {'class': 'event-title'} %}
+            {{ utility.link_if(
+              event.venue,
+              none if compact else (event.data.url or event.url)
+            ) }}
+          {% endh %}
+
+          {% if not event.is_future %}
+            {{ links(event) }}
+          {% elif event.data.discount and not compact %}
+            <p class="event-discount">
+              Use discount code
+              <code>{{ event.data.discount.code }}</code>
+              to get <strong>{{ event.data.discount.amount }} off</strong>
+            </p>
+          {% endif %}
+        </li>
+      {%- endfor -%}
+    </ol>
+  {%- endif -%}
+{% endmacro %}
+
+{# @docs
+label: meta
+note: Display date and location for an event
+params:
+  event:
+    type: object
+    note: |
+      A single event,
+      built by either `getEvents` or `pageEvents`.
+#}
+{% macro meta(event) %}
+  <p class="event-meta">
+    {%- if event.is_future -%}
+      <strong>{{ utility.datetime(event.date) }}</strong>
+    {%- else -%}
+      {{ utility.datetime(event.date) }}
+    {%- endif %}
+    {% if event.data.adr %}
+      {{ ['--', event.data.adr] | join(' ') | mdInline | safe }}
+    {% endif %}
+  </p>
+{% endmacro %}
+
+{# @docs
+label: links
+note: Display event links to slides or video
+params:
+  event:
+    type: object
+    note: |
+      A single event,
+      built by either `getEvents` or `pageEvents`.
+#}
+{% macro links(event) %}
+  {%- set links = [
+    {
+      'title': 'Slides',
+      'url': (event.data.slides or event.page.slides)
+    },
+    {
+      'title': 'Video',
+      'url': event.data.video
+    }
+  ] | selectattr('url') -%}
+
+  {% if links | length > 0 %}
+    <ul class="event-links" inline-list="comma">
+      {%- for link in links -%}
+        <li class="event-link">
+          {{- utility.link_if(link.title, link.url) -}}
```

Comment:
Do you know what best-practice is these days? Add sr-only text to the link, or aria-describedby connection to the heading, or something else?

Reply:
Removing these links from the compact view, since we can't have links in cards. But we should still solve this in the expanded view.

Reply:
@mirisuzanne The expanded view is a little better since when you tab through, it reads the title of the event followed by link to slide and link to video. Since the cards didn't have the title of the event as a link (only the talk topic with the authors name after topic), it wasn't clear which slides from which event you were looking at. I think we can add the name of the conference to the screen reader text before the first link if there are two, or maybe on both. I never know what is too verbose.

Reply:
Added a card for this: https://trello.com/c/g96PXZhv/201-i-want-screen-reader-friendly-links-to-event-slides-and-videos-in-cards-and-on-detail